scripts/createTags.js

```diff
@@ -17,15 +22,47 @@ function main(solutionsDir, outputFile) {
                 var line = lines[i];
                 if(line.startsWith("//")){
                     var parts = line.replace("//","").split("=", 2);
+                    var duplicatePos = tagCompare(tagsWithoutDup, parts[0]);
+                    if(duplicatePos > -1){
+                        if(!frequence.has(parts[0])){
+                            frequence.set(parts[0],1)
+                            parts[0] = parts[0].toLowerCase().replace(/\s+/g,"");
+                        }
+                        else{
+                            frequence.set(parts[0],frequence.get(parts[0]) + 1);
+                            parts[0] = parts[0].toLowerCase().replace(/\s+/g,"");
+                        }
+                    }
+                    else{
+                        frequence.set(parts[0],1);
+                        tagsWithoutDup.push(parts[0].toLowerCase().replace(/\s+/g,""));
+                        parts[0] = parts[0].toLowerCase().replace(/\s+/g,"");         
+                    }
                     if(parts.length == 2){
                         var tagValues = parts[1].split(";");
                         tagValues.forEach(tagValue => {
+                            duplicatePos = tagCompare(tagsWithoutDup, tagValue);
```

Comment:
These blocks look very similar. Can you move them or parts of them to a function?